src/components/CompareResults/SubtestsResults/SubtestsResultsMain.tsx

```diff
@@ -68,102 +64,107 @@ function SubtestsResultsMain({ view }: SubtestsResultsMainProps) {
-        subtestsHeader.base_repo,
-        subtestsHeader.base_rev,
-        subtestsHeader.base_repo,
-        subtestsHeader.new_rev,
-        subtestsHeader.framework_id,
-        subtestsHeader.base_parent_signature,
-        subtestsHeader.new_parent_signature,
-      );
-  }
-
   return (
-    <Container
-      maxWidth={false}
-      sx={{ maxWidth: '1300px' }}
-      className={styles.container}
-      data-testid='subtests-main'
-    >
+    <Container className={styles.container} data-testid='subtests-main'>
       <header>
         <SubtestsBreadcrumbs view={view} />
-        <Alert severity='info' className={styles.title}>
-          A Perfherder link is available for{' '}
-          <Link href={subtestsViewPerfherderURL} target='_blank'>
-            the same results
-          </Link>
-          {'.'}
-        </Alert>
-        <SubtestsRevisionHeader header={subtestsHeader} view={view} />
-        <Grid container spacing={1}>
-          <Grid item xs={12} md={6} sx={{ marginInlineEnd: 'auto' }}>
-            <SearchInput
-              defaultValue={initialSearchTerm}
-              onChange={onSearchTermChange}
-            />
-          </Grid>
-          <Grid item xs='auto'>
-            <DownloadButton resultsPromise={[results]} />
-          </Grid>
-          <Grid item xs='auto'>
-            <RetriggerButton result={results[0]} variant='text' />
-          </Grid>
-        </Grid>
+
+        <Suspense
+          fallback={
+            <div>
+              <Stack spacing={1} sx={{ marginBottom: '12px' }}>
+                <Alert severity='info' className={styles.title}>
+                  A Perfherder link is available for{' '}
+                  <Link href='' target='_blank'>
+                    the same results
+                  </Link>
+                  {'.'}
+                </Alert>
+                <Skeleton
+                  variant='rounded'
+                  sx={{
+                    fontSize: '1.784rem',
+                    backgroundColor:
+                      themeMode === 'light'
+                        ? Colors.SecondaryDefault
+                        : Colors.Background300Dark,
+                  }}
+                  animation='pulse'
+                />
+              </Stack>
+              <Grid container spacing={1}>
+                <Grid item xs={12} md={6} sx={{ marginInlineEnd: 'auto' }}>
+                  <SearchInput
+                    defaultValue={initialSearchTerm}
+                    onChange={onSearchTermChange}
+                  />
+                </Grid>
+                <Grid item xs='auto'>
+                  <DownloadButton resultsPromise={[]} />
+                </Grid>
+                <Grid item xs='auto'>
+                  <RetriggerButton variant='text' />
+                </Grid>
+              </Grid>
+            </div>
+          }
+        >
+          <Await resolve={results}>
+            {(loadedResults: CompareResultsItem[]) => {
+              {
```

Comment:
why 2 blocks??

Reply:
It wuold be better to extract this component, along with all the related logic, to another component (in the same file).

I think the new component could be called something such as `SubtestsResultsMainImpl` or `SubtestsResultsMainInternal`.